### PR TITLE
Render only a selection of pages

### PR DESCRIPTION
**Pull request by t29mato**
## Original issue

https://github.com/lfoppiano/streamlit-pdf-viewer/issues/22

### CODE REVIEW

**Comment by lfoppiano** [comment]
@t29mato I'm not so sure, but it seems you linked the wrong issue  🤔

File: streamlit_app.py [hunk 1]
```diff
@@ -58,7 +58,12 @@
     highlight_figures = st.toggle('Figures and tables', value=True, disabled=not st.session_state['uploaded'])
     highlight_callout = st.toggle('References citations in text', value=True, disabled=not st.session_state['uploaded'])
     highlight_citations = st.toggle('Citations', value=True, disabled=not st.session_state['uploaded'])
-    st.divider()
+
+    st.header("Page Selection")
+    page_options = list(range(1, 101))  # Replace 101 with the actual number of pages + 1
```
Comment (lfoppiano) [inline streamlit_app.py]: I think this line indicates the problem with this feature... 🤔

**Comment by t29mato** [comment]
> @t29mato I'm not so sure, but it seems you linked the wrong issue 🤔

Thank you for letting me know.
Yes, it's supposed to be https://github.com/lfoppiano/streamlit-pdf-viewer/issues/22
I've updated it.

File: streamlit_app.py (hunk 1 above)
Comment (t29mato) [reply streamlit_app.py]: I agree with your point. However, as a parameter for the library, it is an Optional List type and can be passed in the form of `[1, 2, 3]`. How best to show this select box in the Structure-viewer? Another idea I'm considering is to allow users to input arrays as strings. Do you have any other ideas?

Comment (lfoppiano) [reply streamlit_app.py]: No ideas, without having the pdf in advance is not possible to guess any page range

**Comment by lfoppiano** [comment]
I've improved the flow, and now the demo allow the selection of pages after the PDF document is loaded. 
We could have a string like like in the printing form which allow also input such as 1-4, 5, and so on, but I guess for the demo is fine like this 😉

**Comment by t29mato** [comment]
Oh, you set it up to automatically fetch the page numbers on the backend. That's great. Thanks.

**Pull request merged**